client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -313,14 +318,28 @@ public void close() {
     }
 
     private final class KeepAliveTask implements Runnable {
+        
+        private final ChannelFutureListener listener = new ChannelFutureListener() {
+            @Override
+            public void operationComplete(ChannelFuture future) throws Exception {
+                recentMessage.set(true);
+                if (!future.isSuccess()) {
+                    future.channel().pipeline().fireExceptionCaught(future.cause());
+                }
+            }
+        };
+        
         @Override
         public void run() {
-            try {
-                if (!recentMessage.getAndSet(false)) {
-                    getChannel().writeAndFlush(new WireCommands.KeepAlive()).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
+            if (recentMessage.getAndSet(false)) {
```

Comment:
These are the two changes in client behavior, right?
- This change implies keep-alive is sent every 20 seconds even if the connection is healthy. 
- If there is a connection stall for more than 20 seconds then the connection is closed.

Reply:
Yes